vendor/bat-native-ads/src/bat/ads/internal/base/strings/string_html_parse_util.cc

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/base/strings/string_html_parse_util.h"
+
+#include <regex>
+
+namespace ads {
+
+std::string FindFirstRegexMatch(const std::string& search_text,
+                                const std::string& rgx_str) {
+  std::string match_str;
+  std::regex rgx(rgx_str);
```

Comment:
We should not use `std::regex`, see `RE2`

Reply:
Updated to use RE2